src/utils.ts

```diff
@@ -35,8 +44,126 @@ export class Utils {
+     * CREDENTIALS_URL_MODE: if JF_URL exists and the rest of the credentials are found in JF_ACCESS_TOKEN / JF_USER + JF_PASSWORD or a new access token will be generated using OpenID-Connect flow
+     */
+    public static assertCliEnvAndReturnCredCollectionMode(): string {
+        for (let configToken of Utils.getConfigTokens()) {
+            let serverObj = JSON.parse(Buffer.from(configToken, 'base64').toString());
+            if (serverObj && serverObj.artifactoryUrl) {
+                // At this mode all credentials are to be found in on of JF_ENV_* environment variable's value
+                return Utils.CREDENTIALS_ENV_MODE;
+            }
+        }
+        if (process.env.JF_URL) {
+            // At this mode all credentials are to be collected from environment variables: JF_USER, JF_PASSWORD, JF_ACCESS_TOKEN
+            return Utils.CREDENTIALS_URL_MODE;
+        }
+        throw new Error("no JF_ENV_* env variable contains a valid url, nor JF_URL. Please provide a valid url to the server in either way")
+    }
+
+    /**
+     * Retrieves server credentials for accessing JFrog's server
+     * searching for existing environment variables such as JF_ACCESS_TOKEN or the combination of JF_USER and JF_PASSWORD.
+     * If neither is found, and if the request and requester are authorized, it generates an access token for the specified JFrog's server using the OpenID Connect mechanism.
+     */
+    public static async getJfrogCredentials(): Promise<JfrogCredentials> {
+        let jfrogCredentials : JfrogCredentials = this.collectJfrogCredentialsFromEnvVars();
+        if (!jfrogCredentials.jfrogUrl) {
+            return jfrogCredentials
+        }
+
+        // If the required credentials, such as the access token or a combination of username and password, are available, the process terminates without triggering the OIDC flow
+        if (jfrogCredentials.accessToken || (jfrogCredentials.username && jfrogCredentials.password)) {
+            return jfrogCredentials;
+        }
+
+        console.log("JF_ACCESS_TOKEN and JF_USER + JF_PASSWORD weren't found. Getting access token using OpenID Connect")
+        const audience: string = core.getInput(Utils.OIDC_AUDIENCE_ARG, { required: false });
+        let jsonWebToken: string | undefined
+        try {
+            console.log("Fetching JSON web token")
+            jsonWebToken = await core.getIDToken(audience);
+        } catch (error: any){
+            throw new Error(`getting openID Connect JSON web token failed: ${error.message}`)
+        }
+
+        try {
+            return await this.getAccessTokenFromJWT(jfrogCredentials, jsonWebToken)
+        } catch (error: any) {
+            throw new Error(`Exchanging JSON web token with an access token failed: ${error.message}`)
+        }
+    }
 
-    public static async addCliToPath() {
+    /**
+     * Gathers JFrog's credentials from environment variables and delivers them in a JfrogCredentials structure
+     * @private
```

Comment:
Please remove the private and add `@returns`.

Same comment for getAccessTokenFromJWT.